### ISSUE TITLE
Automated documentation system

### ISSUE THREAD

**Issue by JCKodel** [framework7io/framework7#2624]
**Is your feature request related to a problem? Please describe.**
After working with 3 big projects using F7, I made a decision to start to use typescript to keep my sanity.
Typescript requires a .d.ts file for intellisense and type checking.
The only source I found for this is incomplete (and is hard to do it manually).
Angular users also requires .d.ts files, as Angular is best served with TypeScript.

I'm pretty sure that the job of documenting the F7 is preety hard as well.

**Describe the solution you'd like**
If we have some form of formatted or easily parsable documentation, it would be possible to auto generate the documentation website and it would be possible for the community to construct an auto .d.ts generator.

JSDoc (http://usejsdoc.org/) has some interesting tags to make it easy, for instance:

```
/**
 * @param {string} somebody - Somebody's name.
 */
function sayHello(somebody) {
    alert('Hello ' + somebody);
}
```

Output example: https://nhnent.github.io/tui.jsdoc-template/latest/FormButton.html

Only that documentation, per-se, would be enough for TypeScript (based on this accepted answer: https://stackoverflow.com/questions/12687779/how-do-you-produce-a-d-ts-typings-definition-file-from-an-existing-javascript)

**Describe alternatives you've considered**
Now, my $f7, $f7router are all marked as `any` (meaning there is no intellisense nor validation - I often have to leave the official documentation opened and search it to find the correct syntax or available options for a prop).

**Comment by nolimits4web** [comment]
I'm not a fan of JSDoc, it will add even more work for maintaining the framework. But the work on TypeScript definitions is already on the way (in `dts` branch here). So it should arrive in next releases

**Comment by JCKodel** [comment]
That's wondeful news

**Comment by nolimits4web** [comment]
By the way @JCKodel if you work with Vue and TS, do you have an example of correct TypeScript definitions for Vue component? So the component props IntelliSense and its instance methods work correctly

**Comment by JCKodel** [comment]
I'm using vue package and it have a vue.d.ts inside the node_modules\types
(along with many other types).

https://github.com/vuejs/vue/tree/dev/types

**Comment by nolimits4web** [comment]
Yeah, as I understand those typings and shims are mainly for Vue library and app scope. I’m wondering how can we provide .d.ts files for each components like for 
https://github.com/framework7io/framework7/blob/master/packages/vue/components/button.js so when we write something like `<f7-button rai...>` we have `raised` IntelliSense autocomplete

It already works for F7-React components, but we can’t figure out .d.ts syntax for Vue components :)

**Comment by JCKodel** [comment]
AFAIK, you can separate components by modules (with the same name we use on
import, in the same folder) (
https://www.typescriptlang.org/docs/handbook/jsx.html)

Perhaps, this could help:
https://github.com/mui-org/material-ui/tree/master/packages/material-ui/src/Badge
This is a React component set for Material UI. This folder has a Badge
component with .d.ts inside.

Not sure if this is what you are looking for (I would personally use Flow,
if Visual Studio had a plugin for it or, better yet, Haxe, but, we must row
with the flow =\)

**Comment by nolimits4web** [comment]
TypeScript definitions landed in 3.3.0
